cpp/test/prims/batched_matrix.cu

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <common/cuml_allocator.hpp>
+#include <cuML.hpp>
+#include <vector>
+#include "matrix/batched_matrix.hpp"
+#include "random/rng.h"
+#include "test_utils.h"
+
+namespace MLCommon {
+namespace Matrix {
+
+template <typename T>
+struct BatchedMatrixInputs {
+  T tolerance;
+};
+
+template <typename T>
+class BatchedMatrixTest
+  : public ::testing::TestWithParam<BatchedMatrixInputs<T>> {
+ protected:
+  void SetUp() override {
+    using std::vector;
+    params = ::testing::TestWithParam<BatchedMatrixInputs<T>>::GetParam();
+
+    //////////////////////////////////////////////////////////////
+    // Reference matrices
+    // NOTE: cublas expects in column major.
+    // 2x2
+    std::vector<T> A = {0.22814838, 0.92204276,
```

Comment:
Testing seems to be performed only on one fixed size? Can we have this test a few more dataset sizes, please?

Reply:
If I can add Eigen as a dependency, then I could easily make bigger examples with less hard coding.

Reply:
Not sure I got you here. What's limiting from not hardcoding these matrices?

Reply:
Nothing, just annoying copy pasting and formatting the results correctly. It would be easier to use Eigen with random matrices and just compare results to a certain tolerance.

Reply:
I see. @dantegd @JohnZed or @cjnolet do you guys have any objection adding Eigen library as another dependency to cuml?

Reply:
So far. There aren't enough use-cases to justify for a SWIPAT. So, agree with you. Let's just keep this as it is.